jira.go

```diff
@@ -71,13 +77,16 @@ func NewClient(httpClient *http.Client, baseURL string) (*Client, error) {
 
 // NewRawRequest creates an API request.
 // A relative URL can be provided in urlStr, in which case it is resolved relative to the baseURL of the Client.
-// Relative URLs should always be specified without a preceding slash.
 // Allows using an optional native io.Reader for sourcing the request body.
 func (c *Client) NewRawRequest(method, urlStr string, body io.Reader) (*http.Request, error) {
 	rel, err := url.Parse(urlStr)
 	if err != nil {
 		return nil, err
 	}
+	// Relative URLs should be specified without a preceding slash since baseURL will have the trailing slash
+	if !rel.IsAbs() {
```

Comment:
I don't think this is necessary.  Relative paths are defined within the library.  This is under our control.

Reply:
Same with the rest of the `LeftTrim` calls

Reply:
Except if clients call `c.NewRequest()` and use the returned request directly in `c.Do()` -- we do this in our usage a few places where we need to make API calls against API v1 (webhook APIs are still there). 

The way that we're constructing the full url does not preclude a client call passing in an absolute URL path here (i.e., it just works as expected).

Reply:
I guess **technically** a call to `TrimLeft` on an absolute url is a no-op so we could remove the check if we want to keep the PR...